EngineBay.Persistence/BaseDatabaseConfiguration.cs

```diff
@@ -133,6 +133,32 @@ public static string GetDatabaseConnectionString()
             throw new ArgumentException($"Invalid {EnvironmentVariableConstants.DATABASECONNECTIONSTRING} configuration.");
         }
 
+        public static bool IsDetailedDatabaseLoggingEnabled()
+        {
+            var loggingLevelEnvironmentVariable = Environment.GetEnvironmentVariable(EnvironmentVariableConstants.LOGGINGLEVEL);
+
+            if (string.IsNullOrEmpty(loggingLevelEnvironmentVariable))
+            {
+                return false;
+            }
+
+            var loggingLevel = (LogLevel)Enum.Parse(typeof(LogLevel), loggingLevelEnvironmentVariable);
+
+            if (!Enum.IsDefined(typeof(LogLevel), loggingLevel) | loggingLevel.ToString().Contains(',', StringComparison.InvariantCulture))
```

Comment:
Checking if an enum is defined after parsing the value doesn't sit right with me. Will you not have an exception thrown before hitting this if the variable's value is not a valid LogLevel?

Reply:
This needs some work. The validation code after the Parse will also throw an exception, but I am not sure this will ever fire. Needs some unit more unit tests. Will log an issue for this